docker/src/main/java/brooklyn/entity/container/docker/DockerHostImpl.java

```diff
@@ -481,6 +506,25 @@ public void deleteLocation() {
     protected void preStart() {
         getDriver().configureSecurityGroups();
 
+        // Save the VLAN id for this machine
+        MachineProvisioningLocation location = getInfrastructure().getDynamicLocation().getProvisioner();
+        if (isJcloudsLocation(location, SoftLayerConstants.SOFTLAYER_PROVIDER_NAME)) {
```

Comment:
Sadly, there's no callback mechanism for failures on the `JcloudsLocationCustomizer` so we probably just have to wait on a lock with timeout. The way the customizer works is there are multiple methods that can be implemented, the ones that interest us are the post-creation call, which passes in a created machine, and the pre-creation call which offers us a `TemplateOptions` to modify. I think I can work with these. I'm also going to default the scope to the application ID if nothing is defined on the location, which seems like the way most people want this to work.